builtin/mainmenu/tab_online.lua

```diff
@@ -169,6 +169,16 @@ local function get_formspec(tabview, name, tabdata)
 			retval = retval .. "style[btn_delete_favorite;padding=6]"
 			retval = retval .. "image_button[" .. (can_view_clients_list and "4.5" or "5") .. ",1.3;0.5,0.5;" ..
 				core.formspec_escape(defaulttexturedir .. "server_favorite_delete.png") .. ";btn_delete_favorite;]"
+		else
+			retval = retval .. "tooltip[btn_add_favorite;" .. fgettext("Add favorite") .. "]"
+			retval = retval .. "style[btn_add_favorite;padding=6]"
+			if not can_view_clients_list then
```

Comment:
As ywang has suggested, you can use a ternary to vary just the X coordinate below.